lib/pavilion/schedulers/plugins/flux.py

```diff
@@ -0,0 +1,291 @@
+# pylint: disable=too-many-lines
+"""The Flux Framework Scheduler Plugin."""
+
+import os
+import time
+from typing import List, Union, Any, Tuple
+
+import yaml_config as yc
+from pavilion import sys_vars
+from pavilion.jobs import Job, JobInfo
+from pavilion.status_file import STATES, TestStatusInfo
+from pavilion.types import NodeInfo, NodeList
+from ..advanced import SchedulerPluginAdvanced
+from ..config import validate_list
+
+
+# Just import flux once
+try:
+    import flux
+    import flux.job
+    import flux.resource
+    from flux.job import JobspecV1
+except ImportError:
+    flux = None
+
+flux_states = [
+    "DEPEND",
+    "SCHED",
+    "RUN",
+    "CLEANUP",
+    "COMPLETED",
+    "FAILED",
+    "CANCELED",
+    "TIMEOUT",
+]
+
+
+class Flux(SchedulerPluginAdvanced):
+    """
+    Schedule tests with Flux!
+    """
+
+    def __init__(self):
+        super().__init__("flux", "Schedules tests via the Flux Framework scheduler.")
+
+    def _get_config_elems(self):
+        elems = [
+            yc.ListElem(
+                name="avail_states",
+                sub_elem=yc.StrElem(),
+                help_text="When looking for immediately available "
+                "nodes, they must be in one of these "
+                "states.",
+            ),
+            yc.ListElem(
+                name="up_states",
+                sub_elem=yc.StrElem(),
+                help_text="When looking for nodes that could be  "
+                "allocated, they must be in one of these "
+                "states.",
+            ),
+        ]
+
+        defaults = {
+            "up_states": flux_states,
+            "avail_states": flux_states,
+        }
+
+        validators = {
+            "up_states": validate_list,
+            "avail_states": validate_list,
+        }
+
+        return elems, validators, defaults
+
+    @classmethod
+    def parse_node_list(cls, node_list) -> NodeList:
+        """
+        Convert a list of strings into a list of nodes
+        """
+        return NodeList(node_list)
```

Comment:
You can just get rid of this, and do this inline. A lot of these functions are Slurm specific. Check the Advanced and base scheduler classes for the methods that must be overridden (they'll all raise NotImplementedError).